transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/scheduling/persistence/SQLiteEventStoreTest.java

```diff
@@ -445,4 +493,200 @@ public void loadActiveContexts_whenTwoContextsWithOneAvailable_shouldReturnIt()
+        .comparingElementsUsing(CLIENT_METRICS_CORRESPONDENCE)
+        .contains(LOG_SOURCE_METRICS_2);
+    assertThat(clientMetrics.getLogSourceMetricsList())
+        .comparingElementsUsing(CLIENT_METRICS_CORRESPONDENCE)
+        .contains(LOG_SOURCE_METRICS_3);
+  }
+
+  @Test
+  public void loadClientMetrics_shouldIncludeCorrectTimeWindow() {
+    final long START_MS = clock.getTime();
+    store.resetClientMetrics();
+    clock.advance(10);
+    final long END_MS = clock.getTime();
+
+    ClientMetrics clientMetrics = store.loadClientMetrics();
+    assertThat(clientMetrics.getWindow().getStartMs()).isEqualTo(START_MS);
+    assertThat(clientMetrics.getWindow().getEndMs()).isEqualTo(END_MS);
+  }
+
+  @Test
+  public void loadClientMetrics_shouldIncludeCorrectAppNameSpace() {
+    store.resetClientMetrics();
+
+    ClientMetrics clientMetrics = store.loadClientMetrics();
+    assertThat(clientMetrics.getAppNamespace()).isEqualTo(context.getPackageName());
+  }
+
+  @Test
+  public void loadClientMetrics_shouldIncludeCorrectGlobalMetrics() {
+    store.resetClientMetrics();
+    store.recordLogEventDropped(EVENT_DROPPED_COUNT_1, REASON_CACHE_FULL, LOG_SOURCE_1);
+
+    ClientMetrics clientMetrics = store.loadClientMetrics();
+    assertThat(clientMetrics.getGlobalMetrics().getStorageMetrics().getCurrentCacheSizeBytes())
+        .isEqualTo(store.getByteSize());
+    assertThat(clientMetrics.getGlobalMetrics().getStorageMetrics().getMaxCacheSizeBytes())
+        .isEqualTo(EventStoreConfig.DEFAULT.getMaxStorageSizeInBytes());
+  }
+
+  @Test
+  public void
+      loadClientMetrics_whenDbIsEmpty_shouldReturnClientMetricsWithEmptyLogSourceMetricsList() {
+    store.resetClientMetrics();
+
+    ClientMetrics clientMetrics = store.loadClientMetrics();
+    assertThat(clientMetrics.getLogSourceMetricsList().size()).isEqualTo(0);
+  }
+
+  @Test
+  public void resetClintAnalyticsMetrics_shouldRemoveAllRowsInLogEventDroppedTable() {
+    store.recordLogEventDropped(EVENT_DROPPED_COUNT_1, REASON_CACHE_FULL, LOG_SOURCE_1);
+    assertThat(isLogEventDroppedTableEmpty()).isFalse();
+
+    store.resetClientMetrics();
+    assertThat(isLogEventDroppedTableEmpty()).isTrue();
+  }
+
+  @Test
+  public void resetClintAnalyticsMetrics_shouldSetLastMetricsUploadMsToCurrentTime() {
+    final long CURRENT_TIME = clock.getTime();
+    store.resetClientMetrics();
+
+    assertThat(store.loadClientMetrics().getWindow().getStartMs()).isEqualTo(CURRENT_TIME);
+  }
+
+  private boolean isLogEventDroppedTableEmpty() {
+    return store.inTransaction(
+        db ->
+            tryWithCursor(
+                db.rawQuery("SELECT 1 FROM log_event_dropped", new String[] {}),
+                cursor -> cursor.getCount() <= 0));
+  }
+
+  private static final Correspondence<LogSourceMetrics, LogSourceMetrics>
+      CLIENT_METRICS_CORRESPONDENCE =
+      Correspondence.from(
+          SQLiteEventStoreTest::compareLogSourceMetricsByFields, "compare by fields");
```

Comment:
I used Fuzzy Truth to compare them by fields, since generated Proto Object doesn't override `equals` method.

Reply:
Yeah, we have a bug for that b/182387176. I wonder if it would be easier to compare them via their encoded representation. i.e. via `Arrays.equals(actual.toByteArray(), expected.toByteArray())`?

This way we won't have to update this correspondence implementation if we ever add new fields. wdyt?

Reply:
Another challenge is that, we used `List` to represent an collection of `LogSourceMetrics`, but write statement doesn't guarantee to write them in order, and we don't care about their order when comparing them, which could cause tests flaky.

Reply:
The order of repeated fields `must` be preserved according to the spec[1]. If it not the case it's a bug, can you double check that it is indeed an issue?

[1] https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules

Reply:
The issue is when we load the data from the db table, which is not necessarily the order we added them.

Reply:
I see, makes sense